### PR TITLE
dev-python/snakemake: add version 8

### PR DESCRIPTION
**Pull request by mschubert**
* add new dependencies
* add python 3.12 compatibility for packages known to work on Arch
* add version 8 of snakemake

### CODE REVIEW

**Comment by mschubert** [comment]
```
dev-python/smart-open
  NonsolvableDepsInStable: version 6.3.0: nonsolvable depset(bdepend) keyword(~amd64) stable profile (default/linux/amd64/23.0) (62 total): solutions: [ dev-python/google-cloud-storage[python_targets_python3_10(-),python_targets_python3_11(-),python_targets_python3_12(-)] ]
  NonsolvableDepsInStable: version 6.3.0: nonsolvable depset(rdepend) keyword(~amd64) stable profile (default/linux/amd64/23.0) (62 total): solutions: [ dev-python/google-cloud-storage[python_targets_python3_10(-),python_targets_python3_11(-),python_targets_python3_12(-)] ]
```

I'm not sure why `dev-python/smart-open` depends on `dev-python/google-cloud-storage`, this should not be a dependency

**Comment by mschubert** [comment]
> This is a very bad reason to add python compat targets, Arch is known to break python packages every year.

With "known to work on Arch" I didn't mean that the packages are present in the repositories, but that I'm using them as a depdency of snakemake running under python 3.12. That setup is working.

Nonetheless, I've checked all packages explicitly for python 3.12 compatibility:

* `dev-python/connection_pool`: no tests present but import works; is already latest version
* `dev-python/reretry`: tests successful; already latest version
* `dev-python/stopit`: tests successful; already latest version
* `dev-python/throttler`: tests successful; already latest version
* `dev-python/datrie`: tests successful; already latest version
* `dev-python/google-cloud-storage`: version bump 2.17.0 is explicitly compatible with python 3.12; tests broken in python 3.11 (2.7.0) and 3.12 -> disabled tests
* `dev-python/smart-open`: one failed test (azure) already in python 3.11 after fixing moto<5 dep + version update needs more packages without ebuilds -> disabled tests

File: dev-python/smart-open/smart-open-6.3.0.ebuild [hunk 1]
```diff
@@ -24,12 +24,13 @@ RDEPEND="
...
-		dev-python/pytest-rerunfailures[${PYTHON_USEDEP}]
-	)
-"
+#BDEPEND="
+#	test? (
+#		<dev-python/moto-5[${PYTHON_USEDEP}]
+#		dev-python/responses[${PYTHON_USEDEP}]
+#		dev-python/pytest-rerunfailures[${PYTHON_USEDEP}]
+#	)
+#"
```
Comment (AndrewAmmerlaan) [inline dev-python/smart-open/smart-open-6.3.0.ebuild:33]: We don't have to comment out the deps, restricting the test phase is sufficient.

**Comment by AndrewAmmerlaan** [comment]
> dev-python/smart-open: one failed test (azure) already in python 3.11 after fixing moto<5 dep + version update needs more packages without ebuilds -> disabled tests

Alright, but sooner or later it will have to be bumped anyway, we can't keep this version forever.

File: dev-python/immutables/immutables-0.20.ebuild [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Copyright 1999-2024 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DISTUTILS_USE_PEP517=setuptools
```
Comment (AndrewAmmerlaan) [inline dev-python/immutables/immutables-0.20.ebuild:6]:
```

* The directory immutables occludes package installed for python3.12.
* The installed package includes additional files:
*
*     immutables/_map.cpython-312-x86_64-linux-gnu.so
*
* For more information on occluded packages, please see:
* https://projects.gentoo.org/python/guide/test.html#importerrors-for-c-extensions
* Python extension modules (*.so) found installed. Please set:
*   DISTUTILS_EXT=1
* ```

File: dev-python/datrie/datrie-0.8.2-r1.ebuild [hunk 3]
```diff
@@ -3,7 +3,7 @@
 
 EAPI=8
 
-PYTHON_COMPAT=( python3_{10..11} )
+PYTHON_COMPAT=( python3_{10..12} )
```
Comment (AndrewAmmerlaan) [inline dev-python/datrie/datrie-0.8.2-r1.ebuild:6]: datrie fails to compile with gcc-14

**Comment by AndrewAmmerlaan** [comment]
Merged with some minor fixes. Thanks :+1:

**Pull request closed without merging**